### PR TITLE
Beautify README

### PR DESCRIPTION
**Pull request by fabibec**
I updated the README.md and reformated it in a way that is hopefully easier to read, since the text is not splited into the individual persons. Now I arranged the text in the order of execution.

@masterYoda8 I hope thats fine for you.
@tsunaBoy  if you want to add something about reformating I would add it to the end.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +30,30 @@ The column is calculated by the operation i % 14. This will also return an integ
...
+
+All the information for the pathfinding algorithm is contained in an array of the Structure tile. A tile contains the information required for one tile of the map, which is referred to as a node in this context.
+The structure tiles contains the following information:
+- visited | has the tile been processed by the algorithm (bool)
+- distance | the current shortest distance to the starting position (uint8_t)
+- prev | the index of the previous tile on the current shortest path as 1D index (uint8_t)
+
+In order to save memory, all indexes are stored as 1D indexes. We have written a function to convert these coordinates into 2D indexes and vice versa. Hence to this we can always use the index that's more convenient for the current calculation.
+
+The are parts of the algoritm are pretty much equal to a "normal" Dijkstra. In order to explain other interesting logic in more detail, a detailed explanation the algorithm is omitted.
```
Comment (tsunaBoy) [inline README.md:45]: "The are part"?

File: README.md [hunk 2]
```diff
@@ -35,6 +30,30 @@ The column is calculated by the operation i % 14. This will also return an integ
...
+
+## Conversion to driving instructions 
+
+To find the shortest path, we first look for the neighbouring tile of the current destination with the shortest distance to the start. Then a route array of that particular distance is constructed. This array contains the coordinates of the tiles from the start (excluded) to the destination (included). This array is used to generate the driving instructions, with the forward function being executed only if a turn is required or if there are no more coordinates to process.
+
+When the robot arrives at the destination, the makeSound() function is called and the robot turns 180 degrees.
+
+For the way back, a similar array is constructed, now containing the coordinates of the start tile. The execution of the motion commands is done in the same way, the only difference being that the array is now processed from the back to the front.
```
Comment (tsunaBoy) [inline README.md:53]: - The execution of the motion commands 'are' done in the ...

File: README.md [hunk 3]
```diff
@@ -35,6 +30,30 @@ The column is calculated by the operation i % 14. This will also return an integ
...
+
+## Conversion to driving instructions 
+
+To find the shortest path, we first look for the neighbouring tile of the current destination with the shortest distance to the start. Then a route array of that particular distance is constructed. This array contains the coordinates of the tiles from the start (excluded) to the destination (included). This array is used to generate the driving instructions, with the forward function being executed only if a turn is required or if there are no more coordinates to process.
+
+When the robot arrives at the destination, the makeSound() function is called and the robot turns 180 degrees.
+
+For the way back, a similar array is constructed, now containing the coordinates of the start tile. The execution of the motion commands is done in the same way, the only difference being that the array is now processed from the back to the front.  
+
+As the robot arrives at the starting index, it is now turned, based of its celestial direction, in a way that the robot is facing south again, in order to be ready for driving to the next target.
```
Comment (tsunaBoy) [inline README.md:55]: - ... it is now turned, based on its celestial ...

File: README.md [hunk 4]
```diff
@@ -119,10 +138,9 @@ The last operation of the else part is to assign the current degree value to the
 
 The same procedure is used for the right motor.
 
+## Displaying the progress
 
-### Quang Thanh Lai (Route Creation &  Driving Controls)
-TODO
-
+Once the map has been converted from the input string to a 2D array and the start and end points have been processed by the system, this information is printed to the display. This view is updated when the robot starts driving to its first target. Once the robot starts moving, the route is only updated when the robot makes a turn. This design choice was made to keep the display logic out of the driving functions. After each update, you can also see the robot's current position and the destination it's heading towards. The same behaviour can be observed when the robot navigates back to its starting position, or the robot travels to the second target.
```
Comment (tsunaBoy) [inline README.md:143]: - ... when the robot navigates back to its starting position, or 'when it' travels to the second target.

**Review by tsunaBoy — COMMENTED** [review]
There are minor typos to correct
(it's commented by me)

**Comment by fabibec** [comment]
I'll change them later, thanks

File: README.md (hunk 1 above)
Comment (fabibec) [reply README.md:45]: Fixed

File: README.md (hunk 2 above)
Comment (fabibec) [reply README.md:53]: is should be fine here since its refering to the execution not the motion commands

File: README.md (hunk 3 above)
Comment (fabibec) [reply README.md:55]: https://www.quora.com/Which-is-grammatically-correct-based-on-based-upon-or-based-off
Based on seems to be alright

File: README.md (hunk 4 above)
Comment (fabibec) [reply README.md:143]: Fixed!

**Review by tsunaBoy — APPROVED** [review]

**Review by masterYoda8 — APPROVED** [review]

**Pull request merged**